api/tests/routes/public/collective/endpoints/simulate_adage_steps/test_bookings.py

```diff
@@ -379,3 +379,61 @@ def cancel_booking(self, client, booking_id, status_code, json_error=None):
         if json_error:
             for key, msg in json_error.items():
                 assert response.json.get(key) == msg
+
+
+class ResetCollectiveBookingTest(PublicAPIRestrictedEnvEndpointHelper):
+    endpoint_url = "/v2/collective/adage_mock/bookings/{booking_id}/pending"
+    endpoint_method = "post"
+    default_path_params = {"booking_id": 1}
+
+    def test_reset_booking_not_linked_to_key(self, auth_client):
+        booking = factories.CollectiveBookingFactory()
+        with assert_attribute_does_not_change(booking, "status"):
+            self.reset_booking(auth_client, booking.id, status_code=404)
+
+    def test_can_reset_pending_booking(self, auth_client, pending_booking):
+        with assert_attribute_does_not_change(pending_booking, "status"):
+            # 1. get api key
+            # 2. get FF
+            # 3. get collective booking (no update triggered since status does not change)
+            booking_id = pending_booking.id
+            with assert_num_queries(3):
+                self.reset_booking(auth_client, booking_id, status_code=204)
+
+    def test_can_reset_confirmed_booking(self, auth_client, confirmed_booking):
+        with assert_attribute_value_changes_to(confirmed_booking, "status", models.CollectiveBookingStatus.PENDING):
+            # 1. get api key
+            # 2. get FF
+            # 3. get collective booking
+            # 4. update booking
+            booking_id = confirmed_booking.id
+            with assert_num_queries(4):
+                self.reset_booking(auth_client, booking_id, status_code=204)
+
+    def test_can_reset_cancelled_booking(self, auth_client, cancelled_booking):
+        with assert_attribute_value_changes_to(cancelled_booking, "status", models.CollectiveBookingStatus.PENDING):
+            # 1. get api key
+            # 2. get FF
+            # 3. get collective booking and its stock
+            # 4. update booking
+            booking_id = cancelled_booking.id
+            with assert_num_queries(4):
+                self.reset_booking(auth_client, booking_id, status_code=204)
+
+    def test_cannot_reset_used_booking(self, auth_client, used_booking):
+        err = {"code": "CANNOT_SET_BACK_USED_BOOKING_TO_PENDING"}
+        with assert_attribute_does_not_change(used_booking, "status"):
+            self.reset_booking(auth_client, used_booking.id, status_code=403, json_error=err)
```

Comment:
Il faut utiliser `assert_request_has_expected_result` une fois que tu auras mergé ta [PR précédente](https://github.com/pass-culture/pass-culture-main/pull/13932)
```suggestion
            self.assert_request_has_expected_result(auth_client, used_booking.id, status_code=403, json_error=err)
```

Reply:
J'ai appelé la méthode dans `reset_booking` à la place, question de lisibilité. Est-ce que cela te convient ?